design/REGO_V2_PROPOSAL.md

```diff
@@ -540,12 +638,30 @@ result. The result type of the for-in is determined by the surrounding braces.
 ```
 // exists one
 [ for e in list if e < 10 ].size() == 1
-// exists
-[ for e in list if e.matches('hello world') ].size() > 1 
-// all
-{ for k, v in map if k.startsWith('shared') && v > 1 }.size() == map.size() 
+
+// exists at least two unique values, note the `{}` surrounding the `for-in`
+// indicates this is the construction of a set from a list.
+{ for e in list if e.startsWith('t') }.size() > 2
 ```
 
+There are also two special operators which can be used to make qualitative
+inferences about the elements within an aggregate type, `all` and `any`. The
+primary difference between these operators and the filter expressions (or
+even standard comprehensions) is that they are accumulator functions with
+the same semantics as the logical `&&` and `||`, meaning these operators
+can absorb errors. 
+
+```
+// Any element exists in this list
+for e in list any e != 'bad_candidate'
```

Comment:
Queries provide "any" out of the box...If we can write comprehensions inside functions then we don't need the quantifier.

```
function anyNotBadCandidates() {
   for e in list if e != "bad_candidate"
}
```

The all quantifier would be nice to have, but I'd like to consider postponing this to v2.1 because we should be able to get away without it...

```
allGoodCandidates = !anyNotGoodCandidates()

function anyNotGoodCandidates() {
   for e in list if e != "good_candidate"
}
```

Reply:
It turns out that `any` and `all` behave differently than other accumulator functions like `if` since they produce a boolean output rather than a materialized value. Thus, they act with the same semantics of an unrolled list of `||` / `&&` statements respectively.

The following statements are not semantically equivalent
```
// produces a list of bad candidates which can be tested
no_goodies = for e in list if e != "good_candidate"
return !no_goodies.isEmpty()

// produces a boolean output about whether a bad candidate exists
for e in list any e != "good_candidate"

// the following is equivalent to the 'any' behavior
list = [a, b, c]
return list[0] != "good_candidate"
    || list[1] != "good_candidate"
    || list[2] != "good_candidate"
```
We haven't discussed it, but I would prefer to keep boolean as the only true / false value and not define truthy values in terms of non-empty / non-default.